plugins/Kaleidoscope-MouseKeys/src/kaleidoscope/plugin/MouseKeys.h

```diff
@@ -41,20 +41,34 @@ class MouseKeys_ : public kaleidoscope::Plugin {
   EventHandlerResult onNameQuery();
   EventHandlerResult afterEachCycle();
   EventHandlerResult onKeyEvent(KeyEvent &event);
+  EventHandlerResult onAddToReport(Key key);
+  EventHandlerResult afterReportingState(const KeyEvent &event);
 
  private:
   static uint16_t move_start_time_;
   static uint16_t accel_start_time_;
   static uint16_t wheel_start_time_;
 
+  // Mouse cursor and wheel movement directions are stored in a single bitfield
+  // to save space.  The low four bits are for cursor movement, and the high
+  // four are for wheel movement.
+  static constexpr uint8_t wheel_offset_ = 4;
+  static constexpr uint8_t wheel_mask_ = static_cast<uint8_t>(uint8_t(~0) << wheel_offset_);
+  static constexpr uint8_t move_mask_ = static_cast<uint8_t>(uint8_t(~0) ^ wheel_mask_);
```

Comment:
In theory, could  this be done more cleanly using a struct?

Reply:
Yeah, I'm not thrilled about all of the ridiculous gymnastics to get these bitmasks defined.  I could just write:
```c++
static constexpr uint8_t wheel_mask_ {0b11110000};
static constexpr uint8_t move_mask_  {0b00001111};
```
…and that would probably be clearer, even though it doesn't use the 4-bit offset defined above to construct them.

I don't see how a struct would help, though.  Did you have something specific in mind?

Reply:
I do not have a stable preference for either making it more unambiguous to humans or making it more rigidly defined for the compiler, so feel free to make the call on this.

Reply:
I tend to land pretty solidly on "making it clear to humans" over rigid definition for the compiler. There are typically only one or two compilers that need to get it right, but plenty of humans who need to deal with the code over its lifecycle.

Reply:
If there was a clearer way of enforcing the relationship via code, with just the one constant (the 4-bit offset), and the other constants derived from that, I would do it.  This is one of those cases where I really wish I could write this in Rust:
```rust
const OFFSET: u8 = 4;
const MOVE_MASK: u8 = u8::MAX >> OFFSET;
const WHEEL_MASK: u8 = !MOVE_MASK;
```
…but C++ makes it much less clear.  I will update momentarily.

Reply:
I forgot to mention that I fixed the bitmask constants to make them more human-readable.